llvm/lib/Analysis/IVDescriptors.cpp

```diff
@@ -1081,15 +1086,15 @@ RecurrenceDescriptor::getReductionOpChain(PHINode *Phi, Loop *L) const {
   // more expensive than out-of-loop reductions, and need to be costed more
   // carefully.
   unsigned ExpectedUses = 1;
-  if (RedOp == Instruction::ICmp || RedOp == Instruction::FCmp)
+  if (IsNonArithmetic)
```

Comment:
I'm not sure this patch is really NFC because in theory `IsNonArithmetic` could be true when Kind is RecurKind::None. Also, even if you remove references to `getOpcode` for AnyOf recurrence kinds in this function there are still lots of places in LoopVectorize.cpp that call RecurrenceDescriptor::getOpcode(). It's not obvious to me right now that you can avoid calling `getOpcode` for AnyOf types. Also, if you have to specialise the code everywhere to avoid calling `getOpcode` for AnyOf kinds wouldn't that make the code even more complicated?